docs: add map and array functions (#1644)
databendlabs/databend-docs: docs/en/sql-reference/20-sql-functions/10-map-functions/map-insert.md

---
title: MAP_INSERT
---
import FunctionDescription from '@site/src/components/FunctionDescription';

<FunctionDescription description="Introduced or updated: v1.2.654"/>

Returns a new MAP consisting of the input MAP with a new key-value pair inserted (an existing key updated with a new value).

## Syntax

```sql
MAP_INSERT( <map>, <key>, <value> [, <updateFlag> ] )
```

## Arguments

| Arguments      | Description                                                                                      |
|----------------|--------------------------------------------------------------------------------------------------|
| `<map>`        | The input MAP.                                                                                   |
| `<key>`        | The new key to insert into the MAP.                                                              |
| `<value>`      | The new value to insert into the MAP.                                                            |

> | `<updateFlag>` | The boolean flag identifies whether can overwrite the key that already exists. default is FALSE. |

Reviewer: ```suggestion
| `<updateFlag>` | The boolean flag indicates whether an existing key can be overwritten. The default is FALSE. |
```
